loopdb/sql_store.go

```diff
@@ -630,6 +752,170 @@ func (s *BaseDB) convertLoopInRow(row sqlc.GetLoopInSwapsRow,
+
+// BatchToUpsertArgs converts a Batch struct to the arguments needed to insert
+// it into the database.
+func (s *BaseDB) BatchToUpdateArgs(
+	batch Batch) sqlc.UpdateBatchParams {
+
+	args := sqlc.UpdateBatchParams{
+		ID:    batch.ID,
+		State: batch.State,
+		BatchTxID: sql.NullString{
+			Valid:  true,
+			String: batch.BatchTxid.String(),
+		},
+		BatchPkScript: batch.BatchPkScript,
+		LastRbfHeight: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfHeight,
+		},
+		LastRbfSatPerKw: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfSatPerKw,
+		},
+	}
+
+	return args
+}
+
+// ConvertSweepRow converts a sweep row from db to a sweep struct.
+func (s *BaseDB) ConvertSweepRow(ctx context.Context,
+	row sqlc.GetBatchSweepsRow) (Sweep, error) {
+
+	sweep := Sweep{
+		ID:      row.ID,
+		BatchID: row.BatchID,
+		Amount:  btcutil.Amount(row.Amt),
+	}
+
+	swapHash, err := lntypes.MakeHash(row.SwapHash)
+	if err != nil {
+		return sweep, err
+	}
+
+	sweep.SwapHash = swapHash
+
+	hash, err := chainhash.NewHash(row.OutpointTxid)
+	if err != nil {
+		return sweep, err
+	}
+
+	sweep.Outpoint = wire.OutPoint{
+		Hash:  *hash,
+		Index: uint32(row.OutpointIndex),
+	}
+
+	updates, err := s.Queries.GetSwapUpdates(ctx, row.SwapHash)
+	if err != nil {
+		return sweep, err
+	}
+
+	sweep.LoopOut, err = s.convertLoopOutRow(
+		sqlc.GetLoopOutSwapRow{
+			ID:                     row.ID,
+			SwapHash:               row.SwapHash,
+			Preimage:               row.Preimage,
+			InitiationTime:         row.InitiationTime,
+			AmountRequested:        row.AmountRequested,
+			CltvExpiry:             row.CltvExpiry,
+			MaxMinerFee:            row.MaxMinerFee,
+			MaxSwapFee:             row.MaxSwapFee,
+			InitiationHeight:       row.InitiationHeight,
+			ProtocolVersion:        row.ProtocolVersion,
+			Label:                  row.Label,
+			SwapHash_2:             row.SwapHash_2,
+			DestAddress:            row.DestAddress,
+			SwapInvoice:            row.SwapInvoice,
+			MaxSwapRoutingFee:      row.MaxSwapRoutingFee,
+			SweepConfTarget:        row.SweepConfTarget,
+			HtlcConfirmations:      row.HtlcConfirmations,
+			OutgoingChanSet:        row.OutgoingChanSet,
+			PrepayInvoice:          row.PrepayInvoice,
+			MaxPrepayRoutingFee:    row.MaxPrepayRoutingFee,
+			PublicationDeadline:    row.PublicationDeadline,
+			IsExternalAddr:         row.IsExternalAddr,
+			SwapHash_3:             row.SwapHash_3,
+			SenderScriptPubkey:     row.SenderScriptPubkey,
+			ReceiverScriptPubkey:   row.ReceiverScriptPubkey,
+			SenderInternalPubkey:   row.SenderInternalPubkey,
+			ReceiverInternalPubkey: row.ReceiverInternalPubkey,
+			ClientKeyFamily:        row.ClientKeyFamily,
+			ClientKeyIndex:         row.ClientKeyIndex,
+		}, updates,
+	)
+
+	return sweep, err
+}
+
+// SweepToUpsertArgs converts a Sweep struct to the arguments needed to insert.
+func (s *BaseDB) SweepToUpsertArgs(
+	sweep Sweep) (sqlc.UpsertSweepParams, error) {
+
+	args := sqlc.UpsertSweepParams{}
```

Comment:
It is much nicer to full the struct inline (and in this case can immediately return it too).